libtransmission/handshake.cc

```diff
@@ -792,6 +792,7 @@ void tr_handshake::on_error(tr_peerIo* io, tr_error const& error, void* vhandsha
             handshake->have_sent_bittorrent_handshake_ = true;
             handshake->set_state(State::AwaitingHandshake);
             io->write_bytes(std::data(msg), std::size(msg), false);
+            return;
```

Comment:
Note: return here and below at 811 to prevent `handshake->done()` from being called because we're not done yet; we're retrying with a different config (e.g. utp -> tcp, encrypted -> unencrypted)